src/main/java/org/spongepowered/common/mixin/core/network/NetHandlerPlayServerMixin.java

```diff
@@ -270,6 +271,13 @@
         return ZERO_LENGTH_ARRAY; // will bypass the for loop after this method.
     }
 
+    @Inject(method = "processUpdateSign", locals = LocalCapture.CAPTURE_FAILHARD,
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/world/WorldServer;notifyBlockUpdate(Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/block/state/IBlockState;Lnet/minecraft/block/state/IBlockState;I)V"))
+    private void impl$setPlayer(CPacketUpdateSign packetIn, final CallbackInfo ci, WorldServer worldserver,
```

Comment:
Why not just redirect `tileEntitySign.markDirty()` and call `tileEntitySign.setPlayer(null)` there? Better than relying on local capture.

Reply:
Done